### PR TITLE
layout for course

### PR DESCRIPTION
**Pull request by chuan97sun**

### CODE REVIEW

**Comment by chuan97sun** [comment]
ready

File: app/controllers/courses_controller.rb [hunk 1]
```diff
@@ -0,0 +1,56 @@
+class CoursesController < ApplicationController
+before_action :set_course, only: [:show, :edit, :update, :destroy]
```
Comment (saobv-2223) [inline app/controllers/courses_controller.rb]:
```suggestion
before_action :set_course, except: %i(index new create)
```

File: app/controllers/courses_controller.rb [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  end
+
+  private 
+
+  def course_params
+    params.require(:course).permit :name, :description, :started_at, :member, :status, :total_month
+  end
+
+  def set_course
+    @course = Course.find params[:id]
```
Comment (saobv-2223) [inline app/controllers/courses_controller.rb]: using find_by instead of find. 
and add case if that course not foubd

File: app/models/course.rb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+class Course < ApplicationRecord
+  enum status: { opening: 0, started: 1, finished:2 }
```
Comment (saobv-2223) [inline app/models/course.rb]:
```suggestion
  enum status: { opening: 0, started: 1, finished: 2 }
```

**Comment by chuan97sun** [comment]
ready

File: app/models/course.rb [hunk 4]
```diff
@@ -0,0 +1,9 @@
+class Course < ApplicationRecord
+  enum status: { opening: 0, started: 1, finished: 2 }
+
+  after_initialize do
+    if self.new_record?
+      self.status ||= :opening
+    end
```
Comment (saobv-2223) [inline app/models/course.rb]:
```suggestion
    self.status ||= :opening if self.new_record?
```

**Comment by chuan97sun** [comment]
ready

File: app/controllers/courses_controller.rb [hunk 5]
```diff
@@ -0,0 +1,60 @@
+class CoursesController < ApplicationController
+before_action :set_course, except: %i(index new create)
```
Comment (saobv-2223) [inline app/controllers/courses_controller.rb]:
```suggestion
  before_action :set_course, except: %i(index new create)
```

File: app/controllers/courses_controller.rb [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+      render :new
+    end
+  end
+
+  def edit; end
+
+  def update
+    if @course.update course_params
+      redirect_to @course
+      flash[:success] =  t ".course_updated"
```
Comment (saobv-2223) [inline app/controllers/courses_controller.rb]:
```suggestion
      flash[:success] = t ".course_updated"
```

File: app/controllers/courses_controller.rb [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+    else
+      flash.now[:danger] =  t ".course_cannot_be_updated"
```
Comment (saobv-2223) [inline app/controllers/courses_controller.rb]:
```suggestion
      flash.now[:danger] = t ".course_cannot_be_updated"
```

**Comment by chuan97sun** [comment]
ready

File: app/controllers/courses_controller.rb [hunk 8]
```diff
@@ -0,0 +1,60 @@
+class CoursesController < ApplicationController
+  before_action :set_course, except: %i(index new create)
+
+  def index
+    @course = Course.order(:name).page params[:page]
```
Comment (saobv-2223) [inline app/controllers/courses_controller.rb]: You should write order to scope in model and call it here

File: app/views/courses/index.html.erb [hunk 9]
```diff
@@ -0,0 +1,12 @@
...
+
+<%= paginate @course %> 
+
+<ul class="courses">
+  <%= render @course %>
+</ul>
+<%= paginate @course %> 
+<% if current_user.admin? %>
+  <%= link_to "create new course", new_course_url %>
+<%end%>
```
Comment (saobv-2223) [inline app/views/courses/index.html.erb]:
```suggestion
  <%= link_to "Create new course", new_course_url %>
<% end %>
```

File: app/controllers/courses_controller.rb (hunk 8 above)
Comment (chuan97sun) [reply app/controllers/courses_controller.rb:5]: i have wrote in model and it worked without adding coding at here

**Comment by chuan97sun** [comment]
ready

**Pull request merged**